### PR TITLE
Have AHRS update its own orientation

### PR DESCRIPTION
**Pull request by peterbarker**
We can have the library do this itself rather than having the vehicles each do it separately.

### CODE REVIEW

**Comment by peterbarker** [comment]
We have an autotest for this.  I also tested with MAVProxy's horizon module, setting AHRS_ORIENTATION to 42 and then back to 0.

File: libraries/AP_AHRS/AP_AHRS_Backend.cpp [hunk 1]
```diff
@@ -70,6 +70,19 @@ void AP_AHRS::add_trim(float roll_in_radians, float pitch_in_radians, bool save_
 void AP_AHRS::update_orientation()
 {
     const enum Rotation orientation = (enum Rotation)_board_orientation.get();
+
+    if (last_orientation == orientation) {
+        // nothing to do
+        return;
+    }
+    // never update while armed - unless we've never updated
```
Comment (tridge) [inline libraries/AP_AHRS/AP_AHRS_Backend.cpp:78]: breaks changes to custom orientations?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_AHRS/AP_AHRS_Backend.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_AHRS/AP_AHRS_Backend.cpp:78]: Fixed.

**Comment by peterbarker** [comment]
This does have a behavioral change.  Like blimp, sub and Copter, Rover, Plane and Tracker won't update the board orientation while armed.

**Comment by tridge** [comment]
what testing has been done?

**Review by tridge — APPROVED** [review]

**Comment by Hwurzburg** [comment]
@peterbarker this does not determine the orientation automatically during cal does it?

**Pull request merged**